src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -278,11 +284,29 @@ public static string ToPlural(this string single, bool inputIsKnownToBeSingular
             return single.Pluralize(inputIsKnownToBeSingular);
         }
 
+        private static string LastWordToPlural(this string single, bool inputIsKnownToBeSingular = true)
+        {
+            var words = single.SplitByCamelCase();
+            var lastWord = words.LastOrDefault();
+            if (lastWord != null)
```

Comment:
How is it possible that we don't have a single word?

Reply:
Added more tests for `SplitByCamelCase` and I believe we will always have a last word. If the input is `_`, the result will be 2 empty strings `{"", ""}`.